### PR TITLE
added documentation for force ssl configuration

### PR DESCRIPTION
**Pull request by senyosimpson**
Still need to proof-read and view locally

### CODE REVIEW

File: reference/services.html.md.erb [hunk 1]
```diff
@@ -57,3 +57,111 @@ If you don't specify handlers, we just forward TCP to your application as is. Th
...
+    [services.ports.http_options.response]
+    header = { myheader = "fly", via = false}
+
+    [services.ports.http_options.https_redirect]
+    enable = true
+    port = 777
+    status_code = 308
+```
+
+### Compress
```
Comment (jsierles) [inline reference/services.html.md.erb:82]: Perhaps better to name this `Disable response compression`

Comment (jsierles) [reply reference/services.html.md.erb:82]: Also, provide example code like all the other options do.

File: reference/services.html.md.erb [hunk 2]
```diff
@@ -57,3 +57,111 @@ If you don't specify handlers, we just forward TCP to your application as is. Th
...
+
+By default, Fly will compress response bodies. You can switch it off by setting the value to `false`.
+
+### Response
```
Comment (jsierles) [inline reference/services.html.md.erb:86]: Same here, maybe `Modifying Response Headers`. And below, "The `response` section allows you to add or remove HTTP headers to the response.

File: reference/services.html.md.erb [hunk 3]
```diff
@@ -57,3 +57,111 @@ If you don't specify handlers, we just forward TCP to your application as is. Th
...
+For multiple value headers
+
+```toml
+[services.ports.http_options.response]
+header = { myheader = ["fly", "io" ] }
+```
+
+#### Remove a header
+
+A header is removed by setting it to `false`
```
Comment (jsierles) [inline reference/services.html.md.erb:108]: Perhaps `Remove a header by setting it 'false'`

File: reference/services.html.md.erb [hunk 4]
```diff
@@ -57,3 +57,111 @@ If you don't specify handlers, we just forward TCP to your application as is. Th
...
+
+```toml
+[services.ports.http_options.response]
+header = { via = false }
+```
+
+### HTTPS redirect
+
+> This option is only applicable to HTTP handlers. If you accidentally set it on an HTTPS handler, nothing will happen
```
Comment (jsierles) [inline reference/services.html.md.erb:117]: Missing . here.

File: reference/services.html.md.erb [hunk 5]
```diff
@@ -57,3 +57,111 @@ If you don't specify handlers, we just forward TCP to your application as is. Th
...
+
+It is also possible to change the status code
+
+```toml
+[services.ports.http_options.https_redirect]
+enable = true
+status_code = 308
+```
+
+which will return a response like so
```
Comment (jsierles) [inline reference/services.html.md.erb:160]: Maybe better not to split up sentences like this. Make a full sentence here like `This configuration responds with:`

File: reference/configuration.html.md [hunk 6]
```diff
@@ -203,12 +203,16 @@ For each external port you want to accept connections on, you will need a `servi
...
+    
+    # optional
+    http_options = { compress = false }
 ```
 
 This example defines an HTTP handler on port 80.
 
 * `handlers` : An array of strings, each string selecting a handler process to terminate the connection with at the edge. Here, the ["HTTP" handler](/docs/services/#http) will accept HTTP traffic and pass it on to the internal port of the application which we defined earlier.
 * `port` : A string which selects which external ports you want Fly to accept traffic on. You can configure an application to listen for global traffic on ports 25, 53, 80, 443, 5000, 8443, 25565, and ports 10000 - 10100.
+* `http_options` : A table that specifies configuration options for HTTP handlers. It is entirely optional and can be left out.
```
Comment (jsierles) [inline reference/configuration.html.md:215]: Maybe just say `An optional table to specify configuration` and cull the second sentence.

File: reference/services.html.md.erb [hunk 7]
```diff
@@ -57,3 +57,111 @@ If you don't specify handlers, we just forward TCP to your application as is. Th
 
 The `proxy_proto` handler adds information about the original connection, including client IP + port and server IP + port (from the client's perspective).  Most applications need additional logic to accept the proxy protocol, either using a prebuilt library or implementing the [proxy protocol](https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt) directly.
 
+
+## _HTTP Options_
+
+`http_options` is used to specify additional configuration for HTTP handlers.
```
Comment (jsierles) [inline reference/services.html.md.erb:63]: Maybe activate with "Use `http_options` to specify..."

**Review by jsierles — CHANGES_REQUESTED** [review]
Thanks for taking care of this. I suggested some changes.

File: reference/services.html.md.erb [hunk 8]
```diff
@@ -57,3 +57,76 @@ If you don't specify handlers, we just forward TCP to your application as is. Th
 
 The `proxy_proto` handler adds information about the original connection, including client IP + port and server IP + port (from the client's perspective).  Most applications need additional logic to accept the proxy protocol, either using a prebuilt library or implementing the [proxy protocol](https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt) directly.
 
+## _Enforcing SSL_
+
+The `force_ssl` configuration option enforces SSL by redirecting HTTP to HTTPS. When enabled, a redirect response is returned with a `301` status code. If a TLS handler is present and this option is set, it will be ignored.
+
+At present, there is no support for redirecting to a TLS handler on a port other than 443, using a different status code or redirecting specific requests.
```
Comment (senyosimpson) [inline reference/services.html.md.erb:64]: @jsierles Should this be included? I'm not sure how we communicate these type of limitations

**Comment by senyosimpson** [comment]
@jsierles I think we should hold off merging this in for a bit of time. At least, let me get it out on the community forum first and then we can put this in

**Comment by senyosimpson** [comment]
The HTTPS redirects feature was moved out of the HTTP options. I'd rather not bundle all the changes together (and I'm not sure we want the HTTP options in the official documentation as of yet)

**Pull request closed without merging**